docs/tutorials/programmatic-upgradeable-module.md

```diff
@@ -0,0 +1,266 @@
+
+- `aptos move build-publish-payload --json-output-file upgrade.json`
+
+If you have placeholder addresses `_` in your `Move.toml` file, run the command here: [Upgrade function payload section](#generate-upgrade-function-payload-with-new-function-added-in-upgraded-module)
+
+
+This will create a json file `upgrade.json`, with the args of the `metadata_serialied` and `code`. Just need to adjust the `function_id` in the json file, change this to:
+
+`<resource_account>::user_info::upgrade`
+
+- Replace `<resource_account>` with your resource account address.
+
+Your `upgrade.json` file should look like:
+
+```json
+{
+  "function_id": "0x090ad1536fe5cfcb5632b3026f99f8415c55b69ce54b6f17ed8cd7edbcb5edfa::user_info::upgrade",
+  "type_args": [],
+  "args": [
+    {
+      "type": "hex",
+      "value": "0x0e436f646544657..."
+    },
+    {
+      "type": "hex",
+      "value": [
+        "0xa11ceb0b060000000a01000c020c1..."
+      ]
+    }
+  ]
+}
+```
+
+1. Now deploy the modified contract:
+
+```zsh
+aptos move run \
+--json-file upgrade.json
+```
+
+Now check the explorer,
+
+1. Search for your resource account address.
+2. Click `Modules`
+
+You should see the source code with the new function `new_function_added` added. It should look like
+```move
+    public entry fun upgrade(
+        owner: &signer,
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+    ) acquires Config {
+        // Get the config we sent in the `init_module` to resource_account
+        let config = borrow_global<Config>(RESOURCE_ACCOUNT);
+        assert!(config.owner == signer::address_of(owner), 1);
+
+        // This is needed to publish/upgrade the contract
+        let signer = account::create_signer_with_capability(&config.signer_cap);
+        code::publish_package_txn(&signer, metadata_serialized, code);
+    }
+
+    public fun new_function_added(): String {
+        string::utf8(b"new_function_added")
+    }
+```
+
+This concludes the tutorial.
+
+## 3. Appendix
+
+### Addresses with Empty Placeholders
+If your [addresses] have empty placeholders, e.g:
+
+```json
+[addresses]
+your_module = "_"
+resource_account = "_"
+```
+
+Run:
+
+`aptos move create-resource-account-and-publish-package --address-name your_module=0x1c3e5574728c3d0d544f5679229fac153c5f7d78b323751b41860c5ffbb21cfd --seed random3 --named-addresses your_module=0x1c3e5574728c3d0d544f5679229fac153c5f7d78b323751b41860c5ffbb21cfd,resource_account=0x090ad1536fe5cfcb5632b3026f99f8415c55b69ce54b6f17ed8cd7edbcb5edfa`
+
+### Generate `upgrade` function payload with new function added in upgraded module
+
+`aptos move build-publish-payload --json-output-file upgrade.json --named-addresses your_module=0x1c3e5574728c3d0d544f5679229fac153c5f7d78b323751b41860c5ffbb21cfd,resource_account=0x090ad1536fe5cfcb5632b3026f99f8415c55b69ce54b6f17ed8cd7edbcb5edfa`
+
```

Comment:
Any full command line commands, I'd do a full backtick so it stands out across the whole page, instead of inlining

```
aptos move build-publish-payload ...
```

instead of

`aptos move build-publish-payload`

It looks a little cleaner and the added bonus is most markdown publishers add a copy button for it.

Reply:
Thanks, have changed the full CLI commands to full backticks.